src/domain/user.repository.interface.ts

```diff
@@ -29,10 +29,32 @@ export default interface UserRepositoryInterface {
    * Returns array of editor tools
    */
   getUserEditorTools: () => EditorTool[];
+
   /**
    * Adds a tool to the user (marketplace mock)
    *
    * @param id - tool id
    */
-  addTool: (id: string) => void;
+  addTool: (id: string) => Promise<EditorTool>;
+
+  /**
+   * Removes a tool from the user (marketplace mock)
+   *
+   * @param id - tool id
+   */
+  removeTool: (id: string) => Promise<EditorTool['id']>;
+
+  /**
+   * Add tool to the store
+   *
+   * @param tool - tool to add
+   */
+  addToolToStore: (tool: EditorTool) => void;
+
+  /**
+   * Remove tool from the store
+   *
+   * @param id - tool id
+   */
+  removeToolFromStore: (id: EditorTool['id']) => void;
```

Comment:
is is a repo internal stuff, should not be in a public interface

Reply:
removed from the interface